litr/src/main/java/com/linkedin/android/litr/thumbnails/ThumbnailExtractParameters.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.thumbnails
+
+import android.graphics.Point
+import android.net.Uri
+import com.linkedin.android.litr.ExperimentalThumbnailsApi
+import com.linkedin.android.litr.render.ThumbnailRenderer
+
+/**
+ * Request parameters for thumbnail extraction.
+ */
+@ExperimentalThumbnailsApi
+data class ThumbnailExtractParameters @JvmOverloads constructor(
+        /**
+         * The URI of the media from which to extract thumbnails.
+         */
+        val mediaUri: Uri,
+        /**
+         * The timestamp, in microseconds, for which to extract the frame. If the frame can't be extracted for a specified timestamp, the nearest frame may be
+         * returned: this frame selection behavior is defined in [mode].
+         */
+        val timestampUs: Long,
+        /**
+         * The renderer to use to process each of the thumbnails before they're returned.
+         *
+         * One instance of the renderer may be shared between requests, as long as the dimensions of the video frames (for media specified in [mediaUri]) and
+         * the [destSize] remain the same.
+         */
+        val renderer: ThumbnailRenderer,
+        /**
+         * A hint to the [ExtractionBehavior] about what approach to use for extracting video frames.
+         */
+        val mode: ExtractionMode = ExtractionMode.Fast,
+        /**
+         * Optional size of the generated thumbnail Bitmap. The video frame will be resized and scaled to fill this [destSize], preserving the video frame's
+         * original aspect ratio.
+         */
+        val destSize: Point? = null,
```

Comment:
Since we are preserving original aspect ratio, passing a resolution (smaller dimension) should be better. What happens when `destSize`'s aspect ratio is different from video's? Or should we just scale the video to fit `destSize` like when we do in transcoding?

Reply:
Let me add this image just so we're using the same language around this
![image](https://user-images.githubusercontent.com/7144594/150618114-35a49c49-4dd4-4081-bfae-1fad2016eadd.png)

> passing a resolution (smaller dimension) should be better

Can you clarify what you mean by resolution? Do you mean we would perform "aspect fit" scaling instead of "aspect fill"?

> What happens when `destSize`'s aspect ratio is different from video's?

Right now it will scale with "aspect fill": the video frame will be resized and cropped to _fill_ the `destSize` area, and the aspect ratio of the video frame is preserved.

Reply:
Actually, do you mean we should do this? -- 
1. the resulting bitmap will always have the same aspect ratio as the source video (i.e. we won't add any whitespace/padding)
2. instead of "destination size", we pass in something like "maximum width" and "maximum height"

Reply:
Originally, I was thinking of doing this:
 - thumbnail bitmap will have the aspect ratio of video
 - by default, thumbnail will also have a resolution of source video (true frame preview)
 - optionally we can pass the "resolution" parameter, thumbnail bitmap's smaller dimension will be equal to resolution. This is closer to how video is defined on Android - resolution + rotation in 90 degree increments. 

For thumbnail use case, it is probably reasonable to aspect fill (AKA center crop) the thumbnail, as long as it is clearly defined in the API.